packages/back-end/src/entities/database.ts

```diff
@@ -0,0 +1,234 @@
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person) WHERE p.name = $name RETURN p",
+        { name }
+      );
+
+      const person = result.records[0].get("p");
+
+      return person;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findFather(person: number | Person): Promise<Person> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)-[:FATHER]->(f:Person) WHERE p.id = $id RETURN f",
+        { id: id }
+      );
+
+      const father = result.records[0].get("f");
+
+      return father;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findMother(person: number | Person): Promise<Person> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)-[:MOTHER]->(m:Person) WHERE p.id = $id RETURN m",
+        { id: id }
+      );
+
+      const mother = result.records[0].get("m");
+
+      return mother;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findPartners(person: number | Person): Promise<Person[]> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)-[:PARTNER]-(partner:Person) WHERE p.id = $id RETURN partner",
+        { id: id }
+      );
+
+      const partners = result.records.map((record) => record.get("partner"));
+
+      return partners;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findChildren(person: number | Person): Promise<Person[]> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)<-[:FATHER|:MOTHER]-(child:Person) WHERE p.id = $id RETURN child",
+        { id: id }
+      );
+
+      const children = result.records.map((record) => record.get("child"));
+
+      return children;
+    }
+    finally {
+      await session.close();
+    }
+  }
+
+  async updatePerson(person: PersonProperties): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "MATCH (p:Person) WHERE p.id = $id SET p = $person",
+        { id: person.id, person }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async createPerson(person: PersonProperties): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "CREATE (p:Person $person)",
+        { person }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async setFather(person: Person, father: Person): Promise<void> {
```

Comment:
Would it be more efficient to create a `setRelatedPerson(person: Person, relatedPerson: Person, relation: Relation)` ?

Same question for a possible `findRelatedPerson` method (instead of findMother, findFather...)